### PR TITLE
Implement burger-menu and game-switching mechanism

### PR DESCRIPTION
**Pull request by Elkasitu**
This feature implies a lot of changes that were necessary in order to
achieve the feature-set desired:

* Dropping of appcompat shenanigans for androidx + material design
* The above implies changing imports and some minimal codebase changes
  in order to achieve the same behaviour as before
* The base layout is now a DrawerLayout, necessary in order to have a
  navigation drawer (burger menu)

As for the feature itself, these are the changes and additions:

* Burger menu with currently two entries, ACNH and ACNL
* Both images and datasets for both games
* Two static tabs for the tabLayout + one created dynamically for ACNL
  (for underwater sea-creatures)
* A mechanism to change games that will load (or reload, to be
  optimized) the respective dataset and images

In the current implementation, a bug exists wherein switching from ACNH
to ACNL will not update the UI immediately (a tab has to be chosen and
then the UI will update) but the opposite does work properly, I blame
Android.

### CODE REVIEW

**Comment by Elkasitu** [comment]
Closes #5

File: app/src/main/java/com/dmonzonis/nookpendium/CollectionActivity.kt [hunk 1]
```diff
@@ -1,37 +1,51 @@
...
-        val parser = RecordXmlParser(this)
-        val recordList: List<Record> = parser.parse(inputStream)
+        navGamesView.setNavigationItemSelectedListener {
+            // FIXME: If another tab other than 0 is selected when changing game, the data
+            // for what normally would be tab 0 will be put into whichever tab is active
+            when (it.itemId) {
+                R.id.miGameAcnh -> changeGame(R.string.game_acnh)
+                R.id.miGameAcnl -> changeGame(R.string.game_acnl)
+            }
+            true
```
Comment (dmonzonis) [inline app/src/main/java/com/dmonzonis/nookpendium/CollectionActivity.kt:37]: ?

File: app/src/main/java/com/dmonzonis/nookpendium/CollectionActivity.kt [hunk 2]
```diff
@@ -42,16 +56,45 @@ class CollectionActivity : AppCompatActivity() {
...
+        if (drawerToggle.onOptionsItemSelected(item)) {
+            return true
+        }
+        return super.onOptionsItemSelected(item)
+    }
+
+    private fun loadGameAssets(tab: TabLayout.Tab?) {
+        val filePath = if (selectedGame == R.string.game_acnh) {
+            if (tabLayout.tabCount > 2) {
+                tabLayout.removeTabAt(2)
```
Comment (dmonzonis) [inline app/src/main/java/com/dmonzonis/nookpendium/CollectionActivity.kt]: Maybe put this logic outside of the variable assignment to make it a bit easier to read, same with the one in the `else`

**Review by dmonzonis — APPROVED** [review]
LGTM, only a couple of small comments

File: app/src/main/java/com/dmonzonis/nookpendium/CollectionActivity.kt (hunk 1 above)
Comment (Elkasitu) [reply app/src/main/java/com/dmonzonis/nookpendium/CollectionActivity.kt:37]: apparently lambda expressions have to return true or some shit like that, if you remove it it won't compile

File: app/src/main/java/com/dmonzonis/nookpendium/CollectionActivity.kt [hunk 3]
```diff
@@ -1,37 +1,51 @@
 package com.dmonzonis.nookpendium
 
 import android.os.Bundle
-import android.support.design.widget.TabLayout
-import android.support.v7.app.AppCompatActivity
-import android.support.v7.widget.LinearLayoutManager
-import android.support.v7.widget.RecyclerView
+import android.view.MenuItem
+import android.widget.Toast
```
Comment (Elkasitu) [inline app/src/main/java/com/dmonzonis/nookpendium/CollectionActivity.kt]: useless

**Pull request merged**